src/core/include/cesium/omniverse/UrlAssetAccessor.h

```diff
@@ -0,0 +1,130 @@
+/* <editor-fold desc="MIT License">
+
+Copyright(c) 2023 Timothy Moore
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+</editor-fold> */
+
+#pragma once
+
+#include "CesiumAsync/AsyncSystem.h"
+#include "CesiumAsync/IAssetAccessor.h"
+
+#include <curl/curl.h>
+
+#include <cstddef>
+#include <filesystem>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+namespace cesium::omniverse {
+// A cache that permits reuse of CURL handles. This is extremely important for performance
+// because libcurl will keep existing connections open if a curl handle is not destroyed
+// ("cleaned up").
+
+struct CurlCache {
+    struct CacheEntry {
+        CacheEntry()
+            : curl(nullptr)
+            , free(false) {}
+        CacheEntry(CURL* in_curl, bool in_free)
```

Comment:
In cesium-native I see the convention `free_` a lot

Reply:
I prefer that too and will make the change.